### ISSUE TITLE
A more efficient way of using goto statement

### ISSUE THREAD

**Issue by wongsingfo** [zpl-c/zpl#37]
The mix of the goto statement and the branch statement is often not a good practice. Because a branch `if ... then A else B` usually implies that exactly one of the `A` or the `B` will be executed. The code in [README.md](https://github.com/zpl-c/zpl/blob/0a0593f711cf3648b2b73f738a29609bbc0afaa4/README.md#example-usage) violates this rule.

https://github.com/zpl-c/zpl/blob/0a0593f711cf3648b2b73f738a29609bbc0afaa4/code/apps/examples/json_benchmark.c#L35

```c++
if (ok) {
  if (!ok2) goto err;
} else {
  err:
}
```

consider to rewrite it as:

```c++
if (! ok) goto err:
if (! ok2) goto err:

err:
```

**Comment by zaklaus** [comment]
Sorry for the late response, thank you for a very good suggestion, I will definitely look into that!

**Comment by zaklaus** [comment]
Fixed in fcbd7c34597a281698127dc6988547faa983c0ec
